examples/kafka-to-pubsub/src/main/java/org/apache/beam/examples/KafkaToPubsub.java

```diff
@@ -0,0 +1,229 @@
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    KafkaToPubsubOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(KafkaToPubsubOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs a pipeline which reads message from Kafka and writes it to GCS.
+   *
+   * @param options arguments to the pipeline
+   */
+  public static PipelineResult run(KafkaToPubsubOptions options) {
+    // Configure Kafka consumer properties
+    Map<String, Object> kafkaConfig = new HashMap<>();
+    Map<String, String> sslConfig = new HashMap<>();
+    if (options.getSecretStoreUrl() != null && options.getVaultToken() != null) {
+      Map<String, Map<String, String>> credentials =
+          getKafkaCredentialsFromVault(options.getSecretStoreUrl(), options.getVaultToken());
+      kafkaConfig = configureKafka(credentials.get(KafkaPubsubConstants.KAFKA_CREDENTIALS));
+    } else {
+      LOG.warn(
+          "No information to retrieve Kafka credentials was provided. "
+              + "Trying to initiate an unauthorized connection.");
+    }
+
+    if (isSslSpecified(options)) {
+      sslConfig.putAll(configureSsl(options));
+    } else {
+      LOG.info(
+          "No information to retrieve SSL certificate was provided by parameters."
+              + "Trying to initiate a plain text connection.");
+    }
+
+    List<String> topicsList = new ArrayList<>(Arrays.asList(options.getInputTopics().split(",")));
+
+    checkArgument(
+        topicsList.size() > 0 && topicsList.get(0).length() > 0,
+        "inputTopics cannot be an empty string.");
+
+    List<String> bootstrapServersList =
+        new ArrayList<>(Arrays.asList(options.getBootstrapServers().split(",")));
+
+    checkArgument(
+        bootstrapServersList.size() > 0 && topicsList.get(0).length() > 0,
+        "bootstrapServers cannot be an empty string.");
+
+    // Create the pipeline
+    Pipeline pipeline = Pipeline.create(options);
+    LOG.info(
+        "Starting Kafka-To-PubSub pipeline with parameters bootstrap servers:"
+            + options.getBootstrapServers()
+            + " input topics: "
+            + options.getInputTopics()
+            + " output pubsub topic: "
+            + options.getOutputTopic());
+
+    /*
+     * Steps:
+     *  1) Read messages in from Kafka
+     *  2) Extract values only
+     *  3) Write successful records to PubSub
+     */
+
+    if (options.getOutputFormat() == FormatTransform.FORMAT.AVRO) {
+      pipeline
+          .apply(
+              "readAvrosFromKafka",
+              FormatTransform.readAvrosFromKafka(
+                  options.getBootstrapServers(), topicsList, kafkaConfig, sslConfig))
+          .apply("createValues", Values.create())
+          .apply("writeAvrosToPubSub", PubsubIO.writeAvros(AvroDataClass.class));
+
+    } else {
```

Comment:
I see why it may seem invaluable for the example, thank you for noticing this! 

I suppose it is worth having the PUBSUB path because this example works out-of-the-box with it. For the AVRO path, the user has to add some code to make it work and also to understand what and how should be changed - the PUBSUB path doesn't require it.

I also updated the README file to highlight the value of it.